app/cli/Main.hs

```diff
@@ -0,0 +1,176 @@
+
+data WalletCommand
+    = WalletCreate WalletCreateOptions
+    | WalletDelete WalletDeleteOptions
+    | WalletUpdate WalletUpdateOptions
+    | WalletList
+    deriving (Eq, Show)
+
+newtype WalletCreateOptions = WalletCreateOptions
+    { name :: WalletName
+    } deriving (Eq, Show)
+
+newtype WalletDeleteOptions = WalletDeleteOptions
+    { id :: WalletId
+    } deriving (Eq, Show)
+
+data WalletUpdateOptions = WalletUpdateOptions
+    { id :: WalletId
+    , name :: WalletName
+    } deriving (Eq, Show)
+
+walletCommand :: Parser Command
+walletCommand = fmap WalletCommand $ subparser $ mconcat
+    [ commandGroup
+        "Wallet commands:"
+    , command "wallet-create" $ info walletCreate $ progDesc
+        "Create a completely new wallet."
+    , command "wallet-delete" $ info walletDelete $ progDesc
+        "Delete an existing wallet."
+    , command "wallet-update" $ info walletUpdate $ progDesc
+        "Update an existing wallet's metadata."
+    , command "wallet-list" $ info walletList $ progDesc
+        "List all existing wallets."
+    ]
+
+walletCreate :: Parser WalletCommand
+walletCreate = WalletCreate . WalletCreateOptions
+    <$> option walletName
+        (  long "name"
+        <> metavar "STRING"
+        <> help "a name for the new wallet" )
+
+walletDelete :: Parser WalletCommand
+walletDelete = WalletDelete . WalletDeleteOptions
+    <$> option walletId
+        (  long "id"
+        <> metavar "UUID"
+        <> help "the ID of the wallet to delete" )
+
+walletUpdate :: Parser WalletCommand
+walletUpdate = fmap WalletUpdate . WalletUpdateOptions
+    <$> option walletId
+        (  long "id"
+        <> metavar "UUID"
+        <> help "the ID of the wallet to update" )
+    <*> option walletName
+        (  long "name"
+        <> metavar "STRING"
+        <> help "a new name for the wallet" )
+
+walletList :: Parser WalletCommand
+walletList = pure WalletList
+
+walletId :: ReadM WalletId
+walletId = eitherReader mkWalletId
+  where
+    mkWalletId :: String -> Either String WalletId
+    mkWalletId = maybe
+        (Left "invalid wallet ID: must be a valid UUID")
+        (pure . WalletId)
+        . UUID.fromString
+
+-- TODO:
+-- Much of the validation logic herein is identical to that defined within
+-- the Cardano.Api.Wallet.Types module. Find a way to reduce the duplication.
+walletName :: ReadM WalletName
+walletName = eitherReader mkWalletName
+  where
+    mkWalletName :: String -> Either String WalletName
+    mkWalletName n
+        | length n < walletNameMinLength =
+            Left $ "name is too short: expected at least "
+                <> show walletNameMinLength <> " chars"
+        | length n > walletNameMaxLength =
+            Left $ "name is too long: expected at most "
+                <> show walletNameMaxLength <> " chars"
+        | otherwise =
+            pure $ WalletName $ T.pack n
+
+    walletNameMinLength = 1
+    walletNameMaxLength = 255
```

Comment:
As the comment indicates, much of this validation logic is the same as that defined within `Cardano.Api.Wallet.Types`. The duplication should probably be removed. Question: where should we place the common validation logic? (I propose locating it along with the type, as a smart constructor.)